autopts/pybtp/btp/tbs.py

```diff
@@ -0,0 +1,216 @@
+    pts_addr_type_get
+from autopts.pybtp.types import BTPError, addr2btp_ba
+
+TBS = {
+    'read_supported_cmds':   (defs.BTP_SERVICE_ID_TBS,
+                              defs.TBS_READ_SUPPORTED_COMMANDS,
+                              CONTROLLER_INDEX, ""),
+    'remote_incoming':       (defs.BTP_SERVICE_ID_TBS,
+                              defs.TBS_REMOTE_INCOMING,
+                              CONTROLLER_INDEX),
+    'set_bearer_name':       (defs.BTP_SERVICE_ID_TBS,
+                              defs.TBS_SET_BEARER_NAME,
+                              CONTROLLER_INDEX),
+    'set_bearer_technology': (defs.BTP_SERVICE_ID_TBS,
+                              defs.TBS_SET_BEARER_TECHNOLOGY,
+                              CONTROLLER_INDEX),
+    'set_uri_scheme_list':   (defs.BTP_SERVICE_ID_TBS,
+                              defs.TBS_SET_URI_SCHEME_LIST,
+                              CONTROLLER_INDEX),
+    'set_status_flags':      (defs.BTP_SERVICE_ID_TBS,
+                              defs.TBS_SET_STATUS_FLAGS,
+                              CONTROLLER_INDEX),
+    'hold_call':             (defs.BTP_SERVICE_ID_TBS,
+                              defs.TBS_HOLD_CALL,
+                              CONTROLLER_INDEX),
+    'remote_hold_call':      (defs.BTP_SERVICE_ID_TBS,
+                              defs.TBS_REMOTE_HOLD_CALL,
+                              CONTROLLER_INDEX),
+    'originate_call':        (defs.BTP_SERVICE_ID_TBS,
+                              defs.TBS_ORIGINATE_CALL,
+                              CONTROLLER_INDEX),
+    'set_signal_strength':   (defs.BTP_SERVICE_ID_TBS,
+                              defs.TBS_SIGNAL_STRENGTH,
+                              CONTROLLER_INDEX),
+}
+
+
+def tbs_command_rsp_succ(timeout=20.0):
+    logging.debug("%s", tbs_command_rsp_succ.__name__)
+
+    iutctl = get_iut()
+
+    tuple_hdr, tuple_data = iutctl.btp_socket.read(timeout)
+    logging.debug("received %r %r", tuple_hdr, tuple_data)
+
+    btp_hdr_check(tuple_hdr, defs.BTP_SERVICE_ID_TBS)
+    return tuple_data
+
+
+def address_to_ba(bd_addr_type=None, bd_addr=None):
+    data = bytearray()
+    bd_addr_ba = addr2btp_ba(pts_addr_get(bd_addr))
+    bd_addr_type_ba = chr(pts_addr_type_get(bd_addr_type)).encode('utf-8')
+    data.extend(bd_addr_type_ba)
+    data.extend(bd_addr_ba)
+    return data
+
+
+def tbs_remote_incoming(index, receiver_uri, caller_uri, friendly_name):
+    logging.debug(f"{tbs_remote_incoming.__name__}")
+
+    data = bytearray()
+    data.extend(struct.pack("b", index))
+    data.extend(struct.pack("b", len(receiver_uri)))
+    data.extend(struct.pack("b", len(caller_uri)))
+    data.extend(struct.pack("b", len(friendly_name)))
+    uri_str = receiver_uri + caller_uri + friendly_name
+    encoded_uri_str = uri_str.encode("utf-8")
+    data.extend(struct.pack("b", len(encoded_uri_str)))
+    fmt = str(len(encoded_uri_str)) + "s"
+    data.extend(struct.pack(fmt, encoded_uri_str))
```

Comment:
I don't know what strings will be used here, but if you take into consideration that:
`len('o'.encode('utf-8'))` gives 1, but
`len('ó'.encode('utf-8'))` gives 2
then maybe it will be safer to encode the strings before retrieving their length

Reply:
Fixed